### PR TITLE
CLI: set JBANG_REPO during surefire tests

### LINKED ISSUE #18224: Release process job still failing on CLI JBang projects
```
Error:  Failures: 
Error:    CliProjectJBangTest.testCreateAppDefaults:47 Expected OK return code. Result:
result: {
  exitCode: {1},
  system_err: {Downloading JBang...
Installing JBang...
[jbang] Resolving dependencies...
[jbang] Loading MavenCoordinate [io.quarkus:quarkus-bom:pom:999-SNAPSHOT]
Error:  [ERROR] Unable to collect/resolve dependency tree for a resolution due to: Could not find artifact io.quarkus:quarkus-bom:pom:999-SNAPSHOT in mavencentral (https://repo1.maven.org/maven2/), caused by: Could not find artifact io.quarkus:quarkus-bom:pom:999-SNAPSHOT in mavencentral (https://repo1.maven.org/maven2/)
[jbang] Run with --verbose for more details
},
  system_out: {}
} ==> expected: <0> but was: <1>
Error:    CliProjectJBangTest.testCreateAppOverrides:77 Expected OK return code. Result:
result: {
  exitCode: {1},
  system_err: {[jbang] Resolving dependencies...
[jbang] Loading MavenCoordinate [io.quarkus:quarkus-bom:pom:999-SNAPSHOT]
Error:  [ERROR] Unable to collect/resolve dependency tree for a resolution due to: Could not find artifact io.quarkus:quarkus-bom:pom:999-SNAPSHOT in mavencentral (https://repo1.maven.org/maven2/), caused by: Could not find artifact io.quarkus:quarkus-bom:pom:999-SNAPSHOT in mavencentral (https://repo1.maven.org/maven2/)
[jbang] Run with --verbose for more details
},
  system_out: {}
} ==> expected: <0> but was: <1>
```

/cc @ebullient @glefloch (you had a look IIRC)

### PR DESCRIPTION
**Pull request by ebullient**
Update the JBANG_REPO environment variable correctly for the release build

Resolves #18224

### CODE REVIEW

**Review by glefloch — COMMENTED** [review]
Nice catch!

File: devtools/cli/src/test/java/io/quarkus/cli/CliDriver.java [hunk 1]
```diff
@@ -311,7 +311,7 @@ public static Result invokeValidateDryRunBuild(Path projectRoot) throws Exceptio
 
     public static Result invokeValidateBuild(Path projectRoot) throws Exception {
         Result result = execute(projectRoot, "build", "-e", "-B", "--clean",
-                "-Dproperty=value1", "-Dproperty2=value2");
```
Comment (maxandersen) [inline devtools/cli/src/test/java/io/quarkus/cli/CliDriver.java:314]: is that change necessary here ?

**Review by maxandersen — APPROVED** [review]
lgtm assuming --verbose is actually needed otherwise just undo that setting.

File: devtools/cli/src/test/java/io/quarkus/cli/CliDriver.java (hunk 1 above)
Comment (ebullient) [reply devtools/cli/src/test/java/io/quarkus/cli/CliDriver.java:314]: Yes. I am verifying the properties go through properly.

Comment (ebullient) [reply devtools/cli/src/test/java/io/quarkus/cli/CliDriver.java:314]: --verbose? not necessarily.. other than we didn't know what was going on ..

**Pull request merged**